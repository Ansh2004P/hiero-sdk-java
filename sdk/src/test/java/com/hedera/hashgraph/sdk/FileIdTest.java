package com.hedera.hashgraph.sdk;

import io.github.jsonSnapshot.SnapshotMatcher;

import com.google.protobuf.InvalidProtocolBufferException;

import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class FileIdTest {
    @BeforeAll
    public static void beforeAll() {
        SnapshotMatcher.start();
    }

    @AfterClass
    public static void afterAll() {
        SnapshotMatcher.validateSnapshots();
    }

    @Test
    void shouldSerializeFromString() {
        SnapshotMatcher.expect(FileId.fromString("0.0.5005").toProtobuf().toString()).toMatchSnapshot();
    }

    @Test
    void toBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(new FileId(5005).toBytes()).toMatchSnapshot();
    }


    @Test
    void fromBytes() throws InvalidProtocolBufferException {
        SnapshotMatcher.expect(FileId.fromBytes(new FileId(5005).toBytes())).toMatchSnapshot();
    }
}
