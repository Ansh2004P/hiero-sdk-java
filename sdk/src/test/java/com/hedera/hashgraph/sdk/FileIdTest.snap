com.hedera.hashgraph.sdk.FileIdTest.shouldSerialize=[
  "# com.hedera.hashgraph.sdk.proto.FileID@bc8a0\nfile_num: 5005\nrealm_num: 0\nshard_num: 0"
]